tests/tck-build-logic/src/main/groovy/org.graalvm.internal.tck-harness.gradle

```diff
@@ -102,6 +103,60 @@ def matrixDefault = [
 
 final String METADATA_GROUP = "Metadata"
 
+List<String> getNewerVersionsFor(String library, String startingVersion) {
+    def baseUrl = "https://repo1.maven.org/maven2"
+    String[] libraryParts = library.split(":")
+    String group = libraryParts[0].replace(".", "/")
+    String artifact = libraryParts[1]
+    def data = new URL(baseUrl + "/" + group + "/" + artifact + "/" + "maven-metadata.xml").getText()
+
+    return tck.getNewerVersionsFromLibraryIndex(data, startingVersion, library)
+}
+
+// gradle getExistingLibrariesWithNewVersions
+Provider<Task> getExistingLibrariesWithNewVersions = tasks.register("getExistingLibrariesWithNewVersions", DefaultTask) { task ->
+    task.setDescription("Returns list of all libraries coordinates")
+    task.setGroup(METADATA_GROUP)
+    task.doFirst {
+        if (!project.hasProperty("libraryMatrixLimit")) {
```

Comment:
:x: Do not access `project` at task execution time. Instead, define a task and declare inputs/outputs.